File: frontend/src/api/worklog.ts
import api from './axios'

export interface WorkLog {
  id: number
  taskId: number
  userId: number
  workDate: string
  timeSpent: number
  remainingEstimate: number
  comment: string
  createdAt: string
}

export interface LogWorkRequest {
  taskId: number
  workDate: string
  timeSpent: number
  remainingEstimate: number
  comment?: string
}

export const workLogApi = {
  getByTask: (taskId: number) => api.get<WorkLog[]>(`/worklogs/task/${taskId}`),
  getMyWorkLogs: () => api.get<WorkLog[]>('/worklogs/my-worklogs'),
  create: (data: LogWorkRequest) => api.post<WorkLog>('/worklogs', data),
}

File: backend/src/main/java/com/burndown/entity/WorkLog.java
package com.burndown.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "work_logs", uniqueConstraints = {
    @UniqueConstraint(columnNames = {"task_id", "user_id", "work_date"})
})
@EntityListeners(AuditingEntityListener.class)
public class WorkLog {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "task_id", nullable = false)
    private Long taskId;

    @Column(name = "user_id", nullable = false)
    private Long userId;

    @Column(name = "work_date", nullable = false)
    private LocalDate workDate;

    @Column(name = "time_spent", nullable = false, precision = 10, scale = 2)
    private BigDecimal timeSpent;

    @Column(name = "remaining_estimate", precision = 10, scale = 2)
    private BigDecimal remainingEstimate;

    @Column(columnDefinition = "TEXT")
    private String comment;

    @CreatedDate
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
